util/LogReader.js

```diff
@@ -0,0 +1,73 @@
+function getNodeMajorVersion() {
+  const v = process.versions.node;
+  const items = v.split(".");
+  if (items.length > 1) {
+    return Number(items[0]);
+  }
+
+  return 0;
+}
+
+let chunksToLinesAsync = null;
+let chomp = null;
+let legacyTail = null;
+let majorVersion = getNodeMajorVersion();
+
+if(majorVersion >= 10) {
+  const stringio = require('../vendor_lib/stringio.js');
+  chomp = stringio.chomp;
+  chunksToLinesAsync = stringio.chunksToLinesAsync;
+} else {
+  legacyTail = require('../vendor_lib/always-tail.js');
+}
+
+const {spawn} = require('child_process');
+const log = require('../net2/logger.js')(__filename);
+
+class Tail {
+  constructor(file) {
+	this.file = file;
+  }
+
+  on(event, callback) {
+	switch(event) {
+    case "line":
+	  this.lineCallback = callback;
+	  break;
+    }
+  }
+
+  async watch() {
+    if(majorVersion < 10) {
+      this.legacyWatch();
+      return;
+    }
+    log.info("Watching file", this.file);
+    const source = spawn("tail",
+                         ["-F", this.file],
+                         { stdio: ['ignore', 'pipe', 'ignore'] }
+                        );
+
+    for await (const line of chunksToLinesAsync(source.stdout)) {
```

Comment:
`for await` is not available on node 8, causes syntax error unless code not required